doc/posts/2023/2023_08_21_vara_week_12_13.rst

```diff
@@ -0,0 +1,55 @@
+Finalized experiments using both datasets: Week 12 & Week13
+============================================================
+
+.. post:: August 21 2023
+   :author: Vara Lakshmi Bayanagari
+   :tags: google
+   :category: gsoc
+
+What I did this week
+~~~~~~~~~~~~~~~~~~~~
+Monai's VQVAE results on T1-weighted NFBS dataset, 125 samples, for batch_size=5 were qualitatively and quantitatively superior to all previous results. I continued the same experiments on the T1-weighted CC359(Calgary-Campinas-359) public dataset consisting of 359 anatomical MRI volumes of healthy individuals. Preprocessed the data using existing `transform_img` function -
+   1. skull-strips the volume using the respective mask
+   2. dipy's `resize` & scipy's `affine_transform` scale the volume to (128,128,128,1) shape
+   3. MinMax normalization to limit the range of intensities to (0,1)
+
+Using existing training parameters, carried out two experiments, one on CC359 alone & another on both datasets combined. Additionally, I made a slight modification in the loss definition by attributing different weights of 0.5 & 1 to background & foreground pixels compared to equal weights from previous experiments. This resulted in faster convergence as shown in the red, blue & purple lines in the combined plot below-
+
+.. image:: /doc/_static/vqvae3d-monai-training-plots.png
+   :alt: Combined trainings plots for all experiments
+   :width: 800
+
+Inference results on the best performing model - B12-both - is as follows-
+
+.. image:: /doc/_static/vqvae-monai-B12-both.png
+   :alt: VQVAE-Monai-B12-both reconstructions & originals showing equally spaced 5 slices for 2 samples
+   :width: 800
+
+This shows that our training not only converged quickly but also improved visually. Here's a comparison of our current best performing model i.e., VQVAE-Monai-B12-both & the previous one i.e., VQVAE-Monai-B5-NFBS. The test reconstruction loss is 0.0013 & 0.0015 respectively.
```

Comment:
I don't think it makes sense to compare losses of models that use different datasets.

Reply:
Both of those mentioned are test losses, it gives an idea of the performance of the model. Think of it as comparing performance between training using 2 different samples, with different lengths, from a data population.

Reply:
If the test losses are from a totally separate dataset, that makes sense. Is it not from CC359 or NFBS?

Reply:
VQVAE-Monai-B12-both uses both datasets(CC359,NFBS)
VQVAE-Monai-B5-NFBS uses only NFBS datasets

Additionally, the test loss for VQVAE-Monai-B10-CC, uses CC359 only, is 0.013 as well. I'll add this result as well. Hope that can help us compare results.

I'll also describe the naming convention properly in the blog.

Reply:
If the test loss is not from a separate dataset
1. We should call it validation loss
2. It does not make sense to compare like this. The training is biased in this case. NFBS trained models has never seen the CC359 training set, while CC359 trained ones have.

Can you clarify this on the blog post?